### PR TITLE
[Do Not Merge]Add O365 Reports Help Docs

### PR DESCRIPTION
**Pull request by falcone01**
Add O365 Reports Help Docs.

### CODE REVIEW

File: api-reference/beta/api/reportroot_emailactivity.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
+# GET: EmailActivity
```
Comment (dkershaw10) [inline api-reference/beta/api/reportroot_emailactivity.md]: I think you should put report in the titles:  Get EmailActivity **Report**

File: api-reference/beta/api/reportroot_emailactivity.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+```http
+GET /reports/EmailActivity(view=view-value, period=period-value, date=date-value)
+```
+
+## Request headers
+
+| Name       | Description|
+|:---------------|:----------|
+| Authorization  | Bearer. required|
```
Comment (dkershaw10) [inline api-reference/beta/api/reportroot_emailactivity.md]: Should be "Bearer &lt;token&gt".  Same for all API topics.

File: api-reference/beta/api/reportroot_emailactivity.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+## Request body
+
+In the request URL, provide following query parameters with values.
+
+| Parameter   | Type|Description|
+|:---------------|:--------|:----------|
+|view|ViewType|View is an enumeration type, used to determine which type of information that current report should return. Can not be null.|
```
Comment (dkershaw10) [inline api-reference/beta/api/reportroot_emailactivity.md]: You should provide all the valid values for this property.

File: api-reference/beta/api/reportroot_emailactivity.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+|period|PeriodType|Period is an enumeration type, used to specify the aggregate type.|
```
Comment (dkershaw10) [inline api-reference/beta/api/reportroot_emailactivity.md]: Again you should specify all the valid values.

File: api-reference/beta/api/reportroot_emailactivity.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+|date|String|A string represented date, format YYYY-MM-DD. Only available when view type is **Detail**. If you call in with date parameter, you will only get back the set of users that did an activity on that day in that product and this data is retained only for 30 days.|
```
Comment (dkershaw10) [inline api-reference/beta/api/reportroot_emailactivity.md]: Last sentence is a little clumsy.  Suggest shortening the whole thing to:
_A string representing date with a format of YYYY-MM-DD. Only available when view type is **Detail** and the date is set within 30 days of the current date. If set, the method returns the set of users that performed an activity on that day._

File: api-reference/beta/api/reportroot_emailactivity.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+
+> Note: When view type is **Detail**, the period parameter will be ignored. For other view types, date parameter will be ignored.
+
+The following **ViewType** are available in this report:
+
+- Detail
+- Activity
+- Users
+
+The following **PeriodType** are available in this report:
```
Comment (dkershaw10) [inline api-reference/beta/api/reportroot_emailactivity.md]: The following **PeriodType** options are available in this report:

File: api-reference/beta/api/reportroot_emailactivity.md [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+
+- D7
+- D30
+- D90
+- D180
+
+> Note: If you call with **Detail** view along with **PeriodType**, the return data is a list of all users that are licensed for the product with their respective last activity date.
```
Comment (dkershaw10) [inline api-reference/beta/api/reportroot_emailactivity.md]: I would merge this note with the last one, so it's all in one place.

File: api-reference/beta/api/reportroot_emailactivity.md [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+
+## Response
+
+If successful, this method returns `200, OK` response code and [Report](../resources/report.md) object in the response body.
```
Comment (dkershaw10) [inline api-reference/beta/api/reportroot_emailactivity.md]: Should say "200 OK" - i.e. no comma AFAIK.

File: api-reference/beta/api/reportroot_emailactivity.md [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+### Request
+
+Here is an example of the request.
+<!-- {
+  "blockType": "request",
+  "name": "reportroot_emailactivity"
+}-->
+
+```http
+GET https://graph.microsoft.com/beta/reports/EmailActivity(view='Detail',period='D7',date=null)
```
Comment (dkershaw10) [inline api-reference/beta/api/reportroot_emailactivity.md]: You don't need to set the date parameter to null explicitly do you?  If not you should removed this parameter from the example request.  If it is required, I would argue this is a bug that should be fixed before GA.

File: api-reference/beta/api/reportroot_emailactivity.md [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+
+### Other valid requests
+
+<!-- {
+  "blockType": "request",
+  "name": "reportroot_emailactivity"
+}-->
+
+```http
+GET https://graph.microsoft.com/beta/reports/EmailActivity(view='Detail',period=null,date='2017-02-02')
```
Comment (dkershaw10) [inline api-reference/beta/api/reportroot_emailactivity.md]: Same comment as before - do you really need to explicitly provide null parameters?

File: api-reference/beta/resources/report.md [hunk 11]
```diff
@@ -0,0 +1,63 @@
+# report resource type
+
+Represent an Office 365 usage report object. It is a stream, which will return a csv file.
```
Comment (dkershaw10) [inline api-reference/beta/resources/report.md]: Represents

File: api-reference/beta/resources/report.md [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+<!-- {
+  "blockType": "resource",
+  "optionalProperties": [
+
+  ],
+  "@odata.type": "microsoft.graph.Report"
+}-->
+
+```csv
+Column1Name,Column2Name
```
Comment (dkershaw10) [inline api-reference/beta/resources/report.md]: I don't think this provides any value to the developer unless you provide a bare minimum CSV report example.

File: api-reference/beta/api/reportroot_emailactivity.md [hunk 13]
```diff
@@ -0,0 +1,111 @@
+# GET: EmailActivity
+
+Retrieve the reports of Exchange Email Activity.
```
Comment (dkershaw10) [inline api-reference/beta/api/reportroot_emailactivity.md]: Retrieve the Exchange Email Activity reports. The response will be a CSV file in a binary stream.

File: api-reference/beta/api/reportroot_emailactivity.md [hunk 14]
```diff
@@ -0,0 +1,111 @@
...
+
+Here is an example of the response. Note: The response object shown here may be truncated for brevity. All of the properties will be returned from an actual call.
+<!-- {
+  "blockType": "response",
+  "truncated": true,
+  "@odata.type": "microsoft.graph.Report"
+} -->
+
+```http
+HTTP/1.1 200 OK
```
Comment (dkershaw10) [inline api-reference/beta/api/reportroot_emailactivity.md]: This doesn't return JSON, so you might want to be clear that what you are showing here would be the "data types" in a CSV file that is actually returned in a binary stream.

File: api-reference/beta/api/reportroot_emailactivity.md (hunk 1 above)
Comment (dkershaw10) [inline api-reference/beta/api/reportroot_emailactivity.md]: NOTE:  Pretty much ALL my comments on this one topic apply to all following API topics.  I've only commented on this topic and the Report resource type.

For the title (applies to all titles) - I think this should have report in the title:  Get EmailActivity report

File: api-reference/beta/api/reportroot_emailactivity.md (hunk 3 above)
Comment (falcone01) [inline api-reference/beta/api/reportroot_emailactivity.md]: I provide this info in the following descriptions

File: api-reference/beta/api/reportroot_emailactivity.md (hunk 4 above)
Comment (falcone01) [inline api-reference/beta/api/reportroot_emailactivity.md]: I provide this info in the following description

**Pull request merged**